### PR TITLE
fix(Table): Prevents from triggering sort when filtering

### LINKED ISSUE #483: Table: clicking on filter toggle triggers sorting
### Describe the bug
Clicking on the filter icon triggers sorting each time.

#### Reproduction
https://itwin.github.io/iTwinUI-react/?path=/story/core-table--filters&args=isSortable:true

#### Additional information
Package version: `@itwin/itwinui-react@1.27.0`

### PR DESCRIPTION
**Pull request by bentleyvk**
Closes #483
I was trying to think of better solution instead of adding `e.stopPropagation()` but every time I would bump into edge case that breaks the fix.
I thought that this should do the trick:
```tsx
<div
  onMouseDown={e => {
    if (e.target === e.currentTarget) {
      onSortClick(e);
    }
  }}
/>  
```
But clicking on sort icon would not sort, also if users have some elements inside header, clicking on them would also not sort.

I am using `userEvent` in `Table` tests because its click simulates real user click: `pointerDown`, `mouseDown`, `focus`, `pointerUp`, `mouseUp`, and `click`.

## Checklist

- [x] Add meaningful unit tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
- [x] ~~Add component features demo in Storybook (different stories)~~
- [x] ~~Approve test images for new stories~~
- [x] ~~Add screenshots of the key elements of the component~~

### CODE REVIEW

File: src/core/Table/Table.test.tsx [hunk 1]
```diff
@@ -806,6 +807,7 @@ it('should not trigger sorting when filter is clicked', () => {
     columns: mockedColumns,
     onFilter,
     onSort,
+    isSortable: true,
```
Comment (bentleyvk) [inline src/core/Table/Table.test.tsx:810]: This was missing therefore test doesn't check anything.

File: src/core/Table/filters/FilterToggle.tsx [hunk 2]
```diff
@@ -56,10 +56,10 @@ export const FilterToggle = <T extends Record<string, unknown>>(
             styleType='borderless'
             isActive={isVisible || column.filterValue}
             className={cx('iui-filter-button', className)}
-            onClick={(e: React.MouseEvent) => {
-              e.stopPropagation();
+            onClick={() => {
               setIsVisible((v) => !v);
             }}
+            onMouseDown={(e) => e.stopPropagation()}
```
Comment (mayank99) [inline src/core/Table/filters/FilterToggle.tsx:62]: Maybe add a comment here? Because it is a bit unintuitive

Comment (bentleyvk) [reply src/core/Table/filters/FilterToggle.tsx:63]: Added.

**Review by mayank99 — APPROVED** [review]

File: src/core/Table/filters/FilterToggle.tsx [hunk 3]
```diff
@@ -56,10 +56,11 @@ export const FilterToggle = <T extends Record<string, unknown>>(
             styleType='borderless'
             isActive={isVisible || column.filterValue}
             className={cx('iui-filter-button', className)}
-            onClick={(e: React.MouseEvent) => {
-              e.stopPropagation();
+            onClick={() => {
```
Comment (veekeys) [inline src/core/Table/filters/FilterToggle.tsx:59]: have you checked if keyboard clicks work fine?

Comment (mayank99) [reply src/core/Table/filters/FilterToggle.tsx:59]: Yes, it does work but we have bigger problems there like not being able to tab into the filter icon (it's hidden without hover						)

Comment (veekeys) [reply src/core/Table/filters/FilterToggle.tsx:59]: it should be visible with focus though...?

Comment (bentleyvk) [reply src/core/Table/filters/FilterToggle.tsx:59]: I was just to write the same. I think I never have tested keyboard navigation for opening filter menu or triggering sort...

**Review by veekeys — APPROVED** [review]

File: src/core/Table/filters/FilterToggle.tsx (hunk 3 above)
Comment (bentleyvk) [reply src/core/Table/filters/FilterToggle.tsx:59]: Created https://github.com/iTwin/iTwinUI-react/issues/488

**Pull request merged**